LoginServer/PipeServer.cs

```diff
@@ -0,0 +1,548 @@
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern SafeFileHandle CreateNamedPipe(
+           String pipeName,
+           uint dwOpenMode,
+           uint dwPipeMode,
+           uint nMaxInstances,
+           uint nOutBufferSize,
+           uint nInBufferSize,
+           uint nDefaultTimeOut,
+           IntPtr lpSecurityAttributes);
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern int ConnectNamedPipe(
+           SafeFileHandle hNamedPipe,
+           IntPtr lpOverlapped);
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern bool DisconnectNamedPipe(SafeFileHandle hHandle);
+
+
+
+        Thread listenThread;
+        readonly List<Client> clients = new List<Client>();
+
+        public class Client
+        {
+            public int id;
+            public static int number = 1;
+            public SafeFileHandle handle;
+            public FileStream stream;
+            public bool isLoggedIn;
+        }
+
+        /// <summary>
+        /// Variable for if the server is currently running.
+        /// </summary>
+        private bool running;
+
+        /// <summary>
+        /// Variable to hold the name of the pipe.
+        /// </summary>
+        private string pipeName;
+
+        /// <summary>
+        /// The size of the stream buffer.
+        /// </summary>
+        public const int BUFFER_SIZE = 4096;
+
+        #region Accessors
+
+        public Thread getListenThread()
+        {
+            return listenThread;
+        }
+
+        /// <summary>
+        /// Returns true if the server is running.
+        /// </summary>
+        /// <returns>bool</returns>
+        public bool IsRunning()
+        {
+            return running;
+        }
+
+        /// <summary>
+        /// Set whether the Server is currently running or not.
+        /// Displays the server is running message.
+        /// </summary>
+        /// <param name="b">Takes a boolean value</param>
+        private void IsRunning(bool b)
+        {
+            running = b;
+        }
+
+        /// <summary>
+        /// Returns the pipe name.
+        /// </summary>
+        /// <returns>string</returns>
+        public string PipeNameIs()
+        {
+            return pipeName;
+        }
+
+        /// <summary>
+        /// Set the name of the pipe.<br></br>
+        /// Takes a string.
+        /// </summary>
+        /// <param name="newPipeName">new name for the pipe</param>
+        private void PipeNameIs(string newPipeName)
+        {
+            pipeName = newPipeName;
+        }
+
+        /// <summary>
+        /// The total number of clients connected to the server.
+        /// </summary>
+        public int TotalConnectedClients
+        {
+            get
+            {
+                lock (clients)
+                {
+                    return clients.Count;
```

Comment:
Probably want clients.Where(c => c.Connected).Count

Reply:
The number of clients in this array, wont necessarily keep up with those that have connected/disconnected.
so by using the where check for all connected clients, you dont overcount.  Just a double certainty.